scripts/lis_hotfix_pipeline/lis-build-vm-power-on-off.ps1

```diff
@@ -0,0 +1,95 @@
+
+    # On/Off
+    [string] $Operation
+)
+
+try {
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/LIS/LISAv2/master/Libraries/TestLogs.psm1 -UseBasicParsing -OutFile ".\TestLogs.psm1"
+    Invoke-WebRequest -Uri https://raw.githubusercontent.com/LIS/LISAv2/master/Libraries/TestHelpers.psm1 -UseBasicParsing -OutFile ".\TestHelpers.psm1"
+    Import-Module ".\scripts\lis_hotfix_pipeline\Library.psm1" -Force
+    Import-Module ".\TestLogs.psm1" -Force
+    Import-Module ".\TestHelpers.psm1" -Force
+
+    Set-Variable -Name LogDir -Value ".\" -Scope Global
+    Set-Variable -Name LogFileName -Value "lis-build-vm-power-on-off.log.txt" -Scope Global
+
+    Register-AzureSubscription($secretsFile)
+
+    $VMs = Get-AzVm | Where-Object {$_.Name -inotmatch 'controller_vm' -and $_.ResourceGroupName -eq $ResourceGroupName} -Verbose
+
+    # Perform the Power ON/OFF operation
+    $Jobs = @()
+    foreach ($VM in $VMs) {
+        if ($Operation -imatch "off") {
+            Write-LogInfo "Stopping $($VM.Name)..."
+            $Jobs += $VM | Stop-AzVm -AsJob -Force
+        } elseif ($Operation -imatch "On") {
+            Write-LogInfo "Starting $($VM.Name)..."
+            $Jobs += $VM | Start-AzVm -AsJob
+        } else {
+            Write-LogErr "Unsupported VM operation '$Operation'. Supported operations : On/Off"
+            exit 1
+        }
+    }
+
+    $InCompleteJobs = ($Jobs | Where-Object {$_.State -imatch "running"}).Count
+    while ($InCompleteJobs -ne 0) {
+        Write-LogInfo "'$InCompleteJobs' jobs still running. Waiting 10 seconds..."
+        Start-Sleep -Seconds 10
+        $InCompleteJobs = 0
+        foreach ($Job in $Jobs) {
+            $JobStatus = Get-Job -Id $Job.Id
+            if ($JobStatus.State -imatch "running") {
+                $InCompleteJobs += 1
+            }
+        }
+    }
+
+    # Remove the jobs once completed.
+    Write-Host "Removing $($Jobs.Count) completed background jobs..."
+    $Jobs | Remove-Job -Force
+
+    # Verify SSH for all VMs...
+    if ($Operation -imatch "On") {
+        $OfflineVMs = 0
+        $Resources = Get-AzResource | Where-Object {$_.ResourceGroupName -eq $ResourceGroupName }
+        $LB = $Resources | Where-Object { $_.ResourceType -eq "Microsoft.Network/loadBalancers" }
+        $LB = Get-AzLoadBalancer -Name $LB.Name -ResourceGroupName $ResourceGroupName
+        $NatRules = $LB.InboundNatRules | Where-Object {$_.BackendIPConfiguration.Id -ne $null }
+        $PublicIP = $Resources | Where-Object { $_.ResourceType -eq "Microsoft.Network/publicIPAddresses" }
+        $PublicIP = Get-AzPublicIpAddress -Name $PublicIP.Name -ResourceGroupName $PublicIP.ResourceGroupName
+        foreach ($Rule in $NatRules) {
+            if (Test-SSH -PublicIP $PublicIP.IpAddress -SSHPort $Rule.FrontendPort) {
+                Write-LogInfo "$($Rule.Name) : VM Online"
+            } else {
+                Write-LogErr "$($Rule.Name) : VM Offline"
+                $OfflineVMs += 1
```

Comment:
Throw the exception from here. Since single offline VM is not acceptable for build.
Additional update for OfflineVMs variable is not required.

Reply:
Update the below comment in code 
```
        # We're iterating through all VMs without breaking out after detecting 1st offline VM.
        # This is done to print all the VMs which were offline and it helps in debugging.
```